components/suggestion/command_provider/app_command_parser.ts

```diff
@@ -610,6 +613,39 @@ export class AppCommandParser {
             return null;
         }
 
+        const values: AppCallValues = parsed.values;
+        parsed.form?.fields.forEach((f) => {
+            if (!values[f.name]) {
+                return;
+            }
+            switch (f.type) {
+            case AppFieldTypes.DYNAMIC_SELECT:
+                values[f.name] = {label: '', value: values[f.name]};
+                break;
+            case AppFieldTypes.STATIC_SELECT:
+                values[f.name] = f.options?.find((o) => (o.value === values[f.name]));
+                break;
+            case AppFieldTypes.USER: {
+                let userName = values[f.name] as string;
+                if (userName[0] === '@') {
+                    userName = userName.substr(1);
+                }
+                const user = getUserByUsername(this.store.getState(), userName);
```

Comment:
What happens when this user has not been loaded from the server yet? e.g. We just refreshed and ran the command as is. Or the username is spelled wrong or is incomplete

Reply:
Current behaviour: an option {label: undefined, value: undefined} would be sent.

What do you think about this approach?
1. Try to get user from redux
2. If not found, try to get user using the getUserByUsername action, which will trigger a call to the server.
3. If not found, somehow surface the error (and return null for this call).